src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,44 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents the Deadline task in the Duke program.
+ */
+public class Deadline extends Task {
+    protected LocalDate by;
+
+    /**
+     * Constructs a Deadline task with the given description.
+     *
+     * @param description Description of the Deadline task.
+     * @param by          Deadline of the Deadline task.
+     */
+    public Deadline(String description, LocalDate by) {
+        super(description);
+        this.by = by;
+    }
+
+    /**
+     * Returns the string representation of this Deadline task to be saved in storage.
+     *
+     * @return the string representation of this Deadline task to be saved in storage.
+     */
+    @Override
+    public String toStringData() {
+        return "D | " + super.toStringData() + " | " + this.by;
```

Comment:
Consider using `String.format` which might be easier to read compared to string concatenation.

Reply:
I agree with @wei2912. The usage of `String.format()` can also be considered in other areas of your code.

Reply:
Hm I see. Thank you for the suggestion!